funnel/forms/account.py

```diff
@@ -35,35 +36,39 @@
 timezones = sorted_timezones()
 
 
-def password_strength_validator(form, field):
-    user_inputs = []
-    if hasattr(form, 'fullname'):
-        user_inputs.append(form.fullname.data)
-    if hasattr(form, 'username'):
-        user_inputs.append(form.username.data)
-    if hasattr(form, 'email'):
-        user_inputs.append(form.email.data)
-    # Test the candidate password
-    tested_password = password_policy.test_password(
-        field.data, user_inputs=user_inputs if user_inputs else None
-    )
-    # Stick password strength into the form for logging in the view and possibly
-    # rendering into UI
-    form.password_strength = float(tested_password['score'])
-    # No test failures? All good then
-    if not tested_password['is_weak']:
-        return
-    # Tell the user to make up a better password
-    raise forms.validators.StopValidation(
-        tested_password['warning']
-        if tested_password['warning']
-        else ' '.join(tested_password['suggestions'])
-        if tested_password['suggestions']
-        else _(
-            "This password is too simple. Add complexity by making it longer and using "
-            "a mix of upper and lower case letters, numbers and symbols"
+class PasswordStrengthValidator:
+    def __init__(self, user_input_fields=[], message=None) -> None:
+        self.user_input_fields = user_input_fields
+        if not message:
+            message = _(
+                "This password is too simple. Add complexity by making it longer and using "
+                "a mix of upper and lower case letters, numbers and symbols"
+            )
+        self.message = message
+
+    def __call__(self, form, field):
+        user_inputs = []
+        for field_name in self.user_input_fields:
+            if hasattr(form, field_name) and getattr(form, field_name) is not None:
+                user_inputs.append(getattr(form, field_name).data)
+
+        tested_password = password_policy.test_password(
+            field.data, user_inputs=user_inputs if user_inputs else None
+        )
+        # Stick password strength into the form for logging in the view and possibly
+        # rendering into UI
+        form.password_strength = float(tested_password['score'])
+        # No test failures? All good then
+        if not tested_password['is_weak']:
+            return
+        # Tell the user to make up a better password
+        raise forms.validators.StopValidation(
+            tested_password['warning']
+            if tested_password['warning']
+            else ' '.join(tested_password['suggestions'])
```

Comment:
IMO `'\n'` or even `'<br>'` (but the whole thing will need a `Markup()` wrapper and each suggestion will need `escape()`).